### PR TITLE
chore(react-spinner): Add a reduced-motion animation for Spinner

### PR DESCRIPTION
**Pull request by behowell**
## Previous Behavior

Spinner was a solid color ring with no animation, when prefers-reduced-motion was set.

## New Behavior

Add a slow gradient-tail animation to Spinner when prefers-reduced-motion is set.

To see a demo:
1. Enable prefers-reduced-motion. 
    * On Windows: go to Settings > Accessibility > Visual effects, set Animation effects = off
3. Open the [Pull-Request Demo Site](https://fluentuipr.z22.web.core.windows.net/pull/30621/public-docsite-v9/storybook/index.html?path=/docs/components-spinner--default)

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/Avatar-mount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/Avatar-mount.html">611</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/Button-mount_base.html">297</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/Button-mount.html">318</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/Field-mount_base.html">1166</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/Field-mount.html">1127</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/FluentProvider-mount_base.html">694</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/FluentProvider-mount.html">714</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/FluentProviderWithTheme-mount_base.html">84</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/FluentProviderWithTheme-mount.html">91</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">66</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">70</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">70</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/MakeStyles-mount_base.html">859</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/MakeStyles-mount.html">850</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/Persona-mount_base.html">1766</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/Persona-mount.html">1692</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/SpinButton-mount_base.html">1350</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30621/perf-test-react-components/SpinButton-mount.html">1409</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr> | `23.517 kB`<br />`8.512 kB` | `23.814 kB`<br />`8.094 kB` | `297 B` <br />`-418 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `70.732 kB`<br />`20.465 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.207 kB`<br />`61.943 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `43.219 kB`<br />`14.299 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `7.944 kB`<br />`2.588 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/9b41bc196072d32e2547c9addf781e3acb03a973'>9b41bc196072d32e2547c9addf781e3acb03a973</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-spinner-65e1fe2a-ca72-408e-90eb-6a1ca3f6caca.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-spinner-65e1fe2a-ca72-408e-90eb-6a1ca3f6caca.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30621&commitId=488f567228db8fc87f41ae7d1e4fd211dcb7e8c1&env=prod&clientType=FLUENTUI)


  <details><summary><b>Spinner converged</b> 15 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Spinner converged.Inverted with Label - High Contrast.chromium.png|203| Changed|
|Spinner converged.Inverted - High Contrast.chromium.png|51| Changed|
|Spinner converged.Inverted - Dark Mode.chromium.png|9| Changed|
|Spinner converged.Primary - High Contrast.chromium.png|18| Changed|
|Spinner converged.Primary with Label - RTL.chromium.png|378| Changed|
|Spinner converged.Primary.chromium.png|8| Changed|
|Spinner converged.Primary with Label - Dark Mode.chromium.png|20| Changed|
|Spinner converged.Primary + size.chromium.png|183| Changed|
|Spinner converged.Primary - Dark Mode.chromium.png|5| Changed|
|Spinner converged.Animation.chromium.png|146| Changed|
|Spinner converged.Inverted with Label.chromium.png|36| Changed|
|Spinner converged.Primary with Label.chromium.png|33| Changed|
|Spinner converged.Inverted.chromium.png|9| Changed|
|Spinner converged.Inverted with Label - Dark Mode.chromium.png|36| Changed|
|Spinner converged.Primary with Label - High Contrast.chromium.png|74| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by behowell** [comment]
Changes were included in #30567

**Pull request closed without merging**